contracts/Vesting.sol

```diff
@@ -145,10 +129,13 @@ contract Vesting is IVesting, Ownable {
                 "Error : 'investors_' or 'amount_' , equal to 0"
             );
             listOfBeneficiaries[investors_[i]].allocationType = allocations_[i];
-            listOfBeneficiaries[investors_[i]].balanceBase += amounts_[i];
             listOfBeneficiaries[investors_[i]].initialReward +=
-                (amounts_[i] / MAX_PERCENTAGE) *
-                _initialPercentage[allocations_[i]];
+                _initialPercentage[allocations_[i]] *
+                (amounts_[i] / MAX_PERCENTAGE);
```

Comment:
1) you need to first multiply and then divide for greater accuracy

Reply:
move  (_initialPercentage[allocations_[i]] *
                    (amounts_[i] / MAX_PERCENTAGE) to separate variable for gas optimization